Nautilus/Patchers/CraftDataPatcher.cs

```diff
@@ -35,45 +35,30 @@ internal static void AddToGroup(TechGroup group, TechCategory category, TechType
             return;
         }
 
-        if(techCategory.Contains(techType))
-        {
-            return;
-        }
+        techCategory.Remove(techType);
 
-        int index = techCategory.IndexOf(after);
+        int index = techCategory.IndexOf(target);
 
         if (index == -1) // Not found
         {
-            techCategory.Add(techType);
-            InternalLogger.Log($"Added \"{techType.AsString():G}\" to groups under \"{group:G}->{category:G}\"", LogLevel.Debug);
+            techCategory.Insert(after ? techCategory.Count : 0, techType);
+            InternalLogger.Log($"{(after ? "Add" : "Insert")}ed \"{techType:G}\" {(after ? "" : "in")}to groups under \"{group:G}->{category:G}\"", LogLevel.Debug);
         }
         else
         {
-            techCategory.Insert(index + 1, techType);
-
-            InternalLogger.Log($"Added \"{techType.AsString():G}\" to groups under \"{group:G}->{category:G}\" after \"{after.AsString():G}\"", LogLevel.Debug);
+            techCategory.Insert(index + (after ? 1 : 0), techType);
+            InternalLogger.Log($"{(after ? "Add" : "Insert")}ed \"{techType:G}\" {(after ? "" : "in")}to groups under \"{group:G}->{category:G}\" {(after ? "after" : "before")} \"{target:G}\"", LogLevel.Debug);
```

Comment:
If we go with an enum for the type of insertion, which I'm still considering for the sake of code clarity, I think this part could be cleaned up a bit. As it stands now it's good as it could be though.